src/shared/path-lookup.c

```diff
@@ -587,14 +586,9 @@ int lookup_paths_init(
                 if (!add)
                         return -ENOMEM;
 
-                if (l) {
-                        r = strv_extend_strv(&l, add, false);
-                        if (r < 0)
+                r = strv_extend_strv(&paths, add, true);
+                if (r < 0)
                                 return r;
-                } else {
-                        l = add;
-                        add = NULL;
-                }
         }
```

Comment:
I don't like this. Now it will always copy `add` into `paths`, even if paths is unset (which it almost always is). Previously, it would just assign `add` to `paths` in this special case, avoiding any copy.